From b2b6d008b090064c9603d319f8f1738634b13da8 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:06:04 -0400
Subject: [tools] perf/lib/traceevent: Do not call warning() directly

Message-id: <1410181752-25631-120-git-send-email-jolsa@redhat.com>
Patchwork-id: 91772
O-Subject: [PATCH RHEL7.1 BZ1134356 119/307] tools lib traceevent: Do not call warning() directly
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 9e9e5dfd86b9fdea8c75bae42cba052901b94f78
Author: Namhyung Kim <namhyung@kernel.org>
Date: Wed Mar 19 10:22:54 2014 +0900

description
===========
The patch 3a3ffa2e82205 ("tools lib traceevent: Report better error
message on bad function args") added the error message but it seems
there's no reason to call warning() directly.

So change it to do_warning_event() to provide event information too.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/event-parse.c b/tools/lib/traceevent/event-parse.c
index 33803c0..baec7d8 100644
--- a/tools/lib/traceevent/event-parse.c
+++ b/tools/lib/traceevent/event-parse.c
@@ -2742,14 +2742,16 @@ process_func_handler(struct event_format *event, struct pevent_function_handler
   type = process_arg(event, farg, &token);
   if (i < (func->nr_args - 1)) {
    if (type != EVENT_DELIM || strcmp(token, ",") != 0) {
-    warning("Error: function '%s()' expects %d arguments but event %s only uses %d",
+    do_warning_event(event,
+     "Error: function '%s()' expects %d arguments but event %s only uses %d",
      func->name, func->nr_args,
      event->name, i + 1);
     goto err;
    }
   } else {
    if (type != EVENT_DELIM || strcmp(token, ")") != 0) {
-    warning("Error: function '%s()' only expects %d arguments but event %s has more",
+    do_warning_event(event,
+     "Error: function '%s()' only expects %d arguments but event %s has more",
      func->name, func->nr_args, event->name);
     goto err;
    }
-- 
1.7.1